### PR TITLE
Update/add packages in the Nextflow ecosystem

### PR DESCRIPTION
**Pull request by marcodelapierre**
This PR adds recipes for 2 new tools in the Nextflow ecosystem:
- [Wave CLI](https://github.com/seqeralabs/wave-cli)
- [Seqera-Kit](https://github.com/seqeralabs/seqera-kit)

It also adds recent versions for the packages Nextflow and Tower CLI.

### CODE REVIEW

**Review by alalazo — CHANGES_REQUESTED** [review]
```
PKG-DIRECTIVES: 1 issue found
1. nf-seqerakit: dependency on py-pyyaml@6.0.0: cannot be satisfied by known versions of py-pyyaml
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/nf-seqerakit/package.py
    known versions of py-pyyaml are 6.0, 5.4.1, 5.3.1, 5.2, 5.1.2, 5.1, 3.13, 3.12, 3.11
```

File: var/spack/repos/builtin/packages/nf-seqerakit/package.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    license("Apache-2.0")
+
+    version("0.4.5", sha256="792bd4fa53de4b3959929413d1ad8f39e20587971c9c5451419da1ff68cf3f49")
+
+    depends_on("nf-tower-cli", type="run")
+
+    depends_on("python@3.8:3", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-pyyaml@6.0.0:", type=("build", "run"))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nf-seqerakit/package.py]: Maybe:
```suggestion
    depends_on("py-pyyaml@6:", type=("build", "run"))
```
?

Comment (marcodelapierre) [reply var/spack/repos/builtin/packages/nf-seqerakit/package.py]: thanks Max!

File: var/spack/repos/builtin/packages/nf-wave-cli/package.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import platform
+
+from spack.package import *
+
+
+class NfWaveCli(Package):
+    """Command line tool for Wave containers provisioning service."""
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nf-wave-cli/package.py:12]: Would it be difficult to build from sources (and have a single version directive for each release)?

Comment (marcodelapierre) [reply var/spack/repos/builtin/packages/nf-wave-cli/package.py:12]: It is a Java application, it requires a JDK and Gradle to compile. 

However, to generate these portable native image executables the JDK requirements  are stricter (I am not an expert). We typically use GraalVM 21.x (https://github.com/oracle/graal).  Adding Spack support for GraalVM seems out of scope (and effort 😅 ) for this PR.   
An alternative approach I could take to have a (still pre-compiled) single recipe is to instead download the JAR file and create a shell script that launches it with the Java runtime.

Though I agree that a single build from source would be nicer,  note the application is not performance critical, it is a utility that can complement the use of Nextflow.

Keen on your thoughts!

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/nf-wave-cli/package.py (hunk 2 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/nf-wave-cli/package.py:12]: Let's revisit this in another PR then :wink:

**Pull request merged**